packages/zowe-explorer/src/utils/AuthUtils.ts

```diff
@@ -61,7 +62,7 @@ export class AuthUtils {
                 await AuthHandler.waitForUnlock(profile);
             } else {
                 // Lock the profile and prompt the user for authentication by providing login/credential prompt options.
-                await AuthHandler.lockProfile(profile, authOpts);
+                await AuthHandler.lockProfile(profile, authOpts as any);
```

Comment:
If `authOpts` should be of type `AuthPromptParams`, so the `as any` cast is not required here

Reply:
the previous code that you were the author of is trying to pass an Error as an ImperativeError causing the type to not match, even using `as unknown as AuthPromptParams` caused build error. not sure why the build error didn't show until I made a change to the file 🤔 
If I set the type in the assignment I get the error, but will look into fixing it.
<img width="621" alt="Screenshot 2025-06-27 at 11 31 54 AM" src="https://github.com/user-attachments/assets/354fb787-8463-4f7e-80c3-d6e69fbc4ee8" />

Reply:
Maybe there's something strange going on with how TypeScript is interpreting the types - as you can see on line 36 in the function we do an `instanceof` check to make sure that `err` is an ImperativeError, then we pass it into `authOpts`. I don't have the build error on my end, not sure what's going on here... 😕 

Since the `instanceof` check has to pass before we build these options, try casting `err as imperative.ImperativeError`, that should have the same effect.

Reply:
What's strange is, my IntelliSense shows `err` is the right type for me:

![image](https://github.com/user-attachments/assets/8176172e-0fc6-4120-8280-1148e3f82875)

And it compiles even if I add `: AuthPromptParams` after `const authOpts`:

![image](https://github.com/user-attachments/assets/4f692a3a-e63d-4750-ae2e-241a568d869c)

Reply:
there are 2 checks now in the check other than instance check, an existing and the one I added
err.message.includes("All configured authentication methods failed"))) ||
            err.message.includes("HTTP(S) status 401")

Reply:
I updated in the comment below, you could consider it a bug that we've always been casting errors as imperative errors (since v3 release), doing an instanceof check is more proper... but it depends on whether you want to consider that a breaking change or just doing a proper check (bug fix)

Either way, thanks for digging into the checks/workarounds